### DISCUSSION TITLE
Ask for some tips to build the source of libcurl.

### DISCUSSION THREAD

**Discussion by babA409866** [curl/curl#11014] [Q&A]
I'm attempting to build a static libcurl.a from source with as many options as possible, but I've found limited guidance on how to do so. The README file only suggests building with './configure --xxxxx', using the desired options; however, it omits many required libraries. For instance, there are no instructions on how to build rtmpdump with curl or which version of OpenSSL to use. Additionally, when I build with more options, other questions arise. Is there a pre-configured project available?

build with options below:

./configure --with-nghttp2 --enable-ldap --with-libssh2 --with-ssl=/usr/local/opt/openssl@3 --with-libidn --with-librtmp --with-zlib --with-brotli

**Comment by babA409866** [comment]
i solve the question . thanks.

**Comment by babA409866** [comment]
i solve the question . thanks.

**Comment by dfandrich** [comment]
Documenting how to build curl statically is a Sisyphean task. See https://curl.se/docs/faq.html#configure_fails_when_using_stati
